runtime/lua/vim/ui.lua

```diff
@@ -29,9 +29,6 @@ local M = {}
 ---               individual item from `items`. Defaults to `tostring`.
 ---     - kind (string|nil)
 ---               Arbitrary hint string indicating the item shape.
----               Plugins reimplementing `vim.ui.select` may wish to
----               use this to infer the structure or semantics of
----               `items`, or the context in which select() was called.
```

Comment:
why remove this note?

Reply:
Because we no longer wish to promote "monkey-patching" API function as an official customization method.

Reply:
where was that discussed? isn't that the purpose of `vim.ui.*` (and was the reason `kind` was introduced)?

Reply:
Pretty much. There was rough consensus in devchat that overloading functions was a bad design choice we shouldn't perpetuate further. I suggested we remove the few instances we document this in order to prevent it from spreading in that case. No one protested so I went ahead with it 🤷.

Want me to keep this?

Reply:
> There was rough consensus in devchat that overloading functions was a bad design choice

Bad choice compare to what? Let's write down the reasoning in an issue. Making these decisions silently in devchat leads to confusion.

Reply:
Bad choice compared to offering an API that allows registering callbacks (since that -- in contrast to monkey-patching -- allows validation).

Reply:
This is totally premature and it's really unsettling that these decisions are not written in an issue. Propose an interface in an issue, don't make these random decisions based on devchat.

Reply:
You're almost making it sound intentionally malicious or something. I didn't think it warranted mentioning as majority of people seemed to seemed to dislike it and it seemed "obvious" at the time that it was a bad idea. I'll remove these changes from there then.